apps/web/components/booking/pages/BookingPage.tsx

```diff
@@ -294,6 +294,46 @@ const BookingPage = ({
   }
 
   const bookEvent = (booking: BookingFormValues) => {
+    const bookingCustomInputs = Object.keys(booking.customInputs || {}).map((inputId) => ({
+      label: eventType.customInputs.find((input) => input.id === parseInt(inputId))?.label || "",
+      value:
+        booking.customInputs && inputId in booking.customInputs && booking.customInputs[inputId]
+          ? booking.customInputs[inputId]
+          : "",
+    }));
+
+    // Check if custom input of type Phone number is valid
+    const customInputsWithPhoneNumberType = eventType.customInputs.filter(
+      (customInput) => customInput.type === EventTypeCustomInputType.PHONE
+    );
+
+    if (customInputsWithPhoneNumberType.length) {
+      let isErrorFound = false;
+      customInputsWithPhoneNumberType.forEach((customInput) => {
```

Comment:
Here we're traversing filtered phone `booking.customInputs` to validate each one.